common/src/lib.rs

```diff
@@ -95,8 +89,17 @@ pub struct BlockHeader {
     pub tx_merkle_root: Hash256,
     /// The Merkle root of the non-essential state.
     pub state_merkle_root: Hash256,
-    /// The essential state.
-    pub essential_state: EssentialState,
+    /// The effective net validator set (delegation-applied) for the next block.
+    /// The order here is the leader order.
+    pub validator_set: Vec<(PublicKey, VotingPower)>,
+    /// The hash of the delegation state stored in the state storage.
+    ///
+    /// This deserves a seperate hash (not integrated in the `state_merkle_root`) due to its special role.
```

Comment:
What do you mean by its special role?

Reply:
Not sure, but I guess it will be frequently (compared to other states) merkle-proved.